gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    try:
+        id_, val_ = vec[0]
```

Comment:
An empty bag-of-words will throw an `IndexError` here, and `isbow` will return False as a result.
I'm not sure if that's a problem or not, but either way, empty input is clearly a corner case we have to be careful about.